### PR TITLE
Make operator / a friend of string_path

### PR DESCRIPTION
**Pull request by ashtum**

### CODE REVIEW

**Comment by ashtum** [comment]
@pdimov, Upon further consideration, I made the `operator/` overload a friend of `string_path` so that it can be implicitly constructed from string and char types when needed. With that, we don't need multiple overloads (the first commit in this PR). Is this the preferred approach?

**Comment by pdimov** [comment]
What are we trying to fix here? https://github.com/boostorg/property_tree/pull/24 and http://trac.cpp.al/trac10/ticket/12466?

**Comment by ashtum** [comment]
> What are we trying to fix here? #24 and http://trac.cpp.al/trac10/ticket/12466?

Yes.

**Comment by pdimov** [comment]
This is the preferred approach, yes. But it looks like we are missing a test for `p1 / p2` with `p1` and `p2` both being paths.

File: test/test_property_tree.hpp [hunk 1]
```diff
@@ -1026,16 +1026,30 @@ void test_path(PTREE *)
...
 
     // Test operator /
+    {
+        path p = path(T("key1")) / path(T("key2.key3"));
+        BOOST_TEST(pt.get<int>(p, 0) == 1);
+    }
+    {
+        path p = path(T("key1.key2")) / path(T("key3"));
+        BOOST_TEST(pt.get<int>(p, 0) == 1);
+    }
```
Comment (ashtum) [inline test/test_property_tree.hpp:1036]: > But it looks like we are missing a test for `p1 / p2` with `p1` and `p2` both being paths.

@pdimov, Addressed.

**Pull request merged**